### PR TITLE
LibVT: Color scheme support

### PR DESCRIPTION
**Pull request by BertalanD**
This PR introduces support for various terminal color schemes. In order to implement it, the following things need to be done:

- [x] Add a `Color` class that lets us store the various color types (RGB, 8-bit indexed color, ANSI named color)
- [x] Add color state to `TerminalWidget`
- [x] Add some color themes to `/res/terminal_colors` (I'm open to suggestions for a nicer name of the directory)
- [x] Add a GUI color scheme switcher to `/bin/Terminal`.

### CODE REVIEW

**Comment by BertalanD** [comment]
Functionally, this should be ready to go, but there is an annoying UI bug. The combo box renders underneath the window.

image

**Comment by BertalanD** [comment]
Can I get a merge on this? 

I'm not sure about why the combo box is buggy, but that's probably not related to any of the code I modified.

Currently, only the default color scheme is included, but I could commit the Solarized and Gruvbox dark themes that I have on my computer. They are under MIT license, so we can pull them in with proper attribution.

**Comment by linusg** [comment]
> Can I get a merge on this?

The title still says WIP, and those PRs usually don't get reviewed. :)

**Comment by BertalanD** [comment]
Can't believe I left it in the title. Thanks for still noticing the comment.

File: Userland/Libraries/LibVT/TerminalWidget.cpp [hunk 1]
```diff
@@ -1135,6 +1140,54 @@ void TerminalWidget::update_paste_action()
     m_paste_action->set_enabled(GUI::Clipboard::the().mime_type().starts_with("text/") && !GUI::Clipboard::the().data().is_empty());
 }
 
+void TerminalWidget::set_color_scheme(const StringView& name)
+{
+    if (name.contains('\\')) {
```
Comment (linusg) [inline Userland/Libraries/LibVT/TerminalWidget.cpp:1145]: Pretty sure you can have a backslash in Unix filenames, this ain't Windows :)
Probably meant to use a forward slash here?

File: Userland/Libraries/LibVT/TerminalWidget.cpp [hunk 2]
```diff
@@ -1135,6 +1140,54 @@ void TerminalWidget::update_paste_action()
...
+        "Red",
+        "Green",
+        "Yellow",
+        "Blue",
+        "Magenta",
+        "Cyan",
+        "White"
+    };
+
+    auto color_config = Core::ConfigFile::open(String::formatted("/res/terminal_colors/{}.ini", name));
```
Comment (linusg) [inline Userland/Libraries/LibVT/TerminalWidget.cpp:1163]: Let's not use an underscore in the path here, that seems pretty uncommon across the system. `terminal-colors` looks less out of place.

**Review by linusg — CHANGES_REQUESTED** [review]
That's pretty cool, nice work!

File: Userland/Libraries/LibVT/Color.h [hunk 3]
```diff
@@ -83,6 +83,17 @@ class Color {
...
 
+    constexpr Color to_bright() const
+    {
+        if (is_named()) {
+            auto numeric_value = static_cast<u16>(as_named());
+            if (numeric_value < 8)
+                return Color::named(static_cast<ANSIColor>(numeric_value + 8));
+            return *this;
+        } else
+            return *this;
```
Comment (linusg) [inline Userland/Libraries/LibVT/Color.h:94]: Please add curly braces to all branches if at least one needs them.

**Pull request merged**